core/modules/shrscan/Director.h

```
@@ -0,0 +1,315 @@
+//! @brief Resume all suspended shared scan queries.
+//!
+//! This method effectively calls resumeQuery() on each suspended query.
+//-----------------------------------------------------------------------------
+
+virtual void  resumeAll() = 0;
+
+//-----------------------------------------------------------------------------
+//! @brief Set the maximum amount of memory that may be locked.
+//!
+//! The default maximum amount of memory that may be locked is equal to largest
+//! amount of memory needed to lock a set of table chunks passed to init().
+//!
+//! @param  bytes  - The maximum number of bytes that may be locked.
+//-----------------------------------------------------------------------------
+
+virtual void  setMaxMemory(unsigned long long bytes) = 0;
+
+//-----------------------------------------------------------------------------
+//! @brief Set the maximum number of queries that be run in parallel.
+//!
+//! The default number of parallel queries is 7.
+//!
+//! @param  maxNum - The maximum number of queries to run in parallel.
+//-----------------------------------------------------------------------------
+
+virtual void  SetMaxQueries(int qnum) = 0;
+
+//-----------------------------------------------------------------------------
+//! @brief Suspend a shared scan query.
+//!
+//! When a running query is suspended, it is placed in the eligible to run
+//! queue in FIFO order. If the query is executing its runQuery() method, the
+//! suspension occurs upon return.
+//!
+//! @param  query   - Pointer to the Query object that describes the query.
+//! @param  unlock  - When true, any additional tables required by the query
+//!                   that were locked are unlocked. Otherwise, the tables
+//!                   remain locked which may prevent new queries to run.
+//!
+//! @return false: The query was not found.
+//! @return true:  The query was suspended or scheduled to be suspended.
```

Should probably raise an exception if query is not found.